framework/src/systems/SystemBase.C

```diff
@@ -1106,6 +1135,15 @@ SystemBase::copyOldSolutions()
     *solutionUDotDotOld() = *solutionUDotDot();
   if (solutionPreviousNewton())
     *solutionPreviousNewton() = *currentSolution();
+
+  if (_solution_state_size > 3)
+  {
+    for (unsigned int i = _solution_state_size - 1; i > 1; --i)
+      *solutionState(i) = *solutionState(i - 1);
+
+    *solutionState(1) = *currentSolution();
```

Comment:
Shouldn't this be...
```suggestion
    *solutionState(1) = *solutionOld();
```

Reply:
Current and old solutions are equal at this point since the solve hasn't been done.

Reply:
@sveerara what do you think?

Reply:
@cbolisetti and @loganharbour, the solution state variable is set to store values in the following format: 
solutionState(0) - current solution at time t
solutionState(1) - solution at t - delta t
solutionState(2) - solution at t- 2*delta t
solutionState(3) - solution at t- 3*delta t ; and so on

So I think when the solution is being shifted backward, solutionState(1) should be set to the solution value at time t, which is current solution. But in the earlier part of this function (line 1102 (old file) or 1132 (edited version)), solutionOld is being set to the currentSolution. So setting solutionState(1) to currentSolution or solutionOld here should return the same result.